app/controllers/timelines_controller.rb

```diff
@@ -0,0 +1,189 @@
+
+  helper_method :access_token_url
+
+  # POST /timelines/paged_index
+  def paged_index
+    # Timelines for index page are loaded dynamically by jquery datatables javascript which
+    # requests the html for only a limited set of rows at a time.
+    records_total = @timelines.count
+    columns = ['title', 'description', 'visibility', 'updated_at', 'tags', 'actions']
+
+    # Filter title
+    title_filter = params['search']['value']
+    @timelines = @timelines.title_like(title_filter) if title_filter.present?
+
+    # Apply tag filter if requested
+    tag_filter = params['columns']['4']['search']['value']
+    @timelines = @timelines.with_tag(tag_filter) if tag_filter.present?
+    timelines_filtered_total = @timelines.count
+
+    sort_column = params['order']['0']['column'].to_i rescue 0
+    sort_direction = params['order']['0']['dir'] rescue 'asc'
+    session[:timeline_sort] = [sort_column, sort_direction]
+    @timelines = @timelines.order(columns[sort_column].downcase => sort_direction)
+    @timelines = @timelines.offset(params['start']).limit(params['length'])
+    response = {
+      "draw": params['draw'],
+      "recordsTotal": records_total,
+      "recordsFiltered": timelines_filtered_total,
+      "data": @timelines.collect do |timeline|
+        copy_button = view_context.button_tag(type: 'button',
+                                              data: { timeline: timeline },
+                                              class: 'copy-timeline-button btn btn-default btn-xs') do
+          "<i class='fa fa-clone' aria-hidden='true'></i> Copy".html_safe
+        end
+        edit_button = view_context.link_to(edit_timeline_path(timeline), class: 'btn btn-default btn-xs') do
+          "<i class='fa fa-edit' aria-hidden='true'></i> Edit Details".html_safe
+        end
+        delete_button = view_context.link_to(timeline_path(timeline), method: :delete, class: 'btn btn-xs btn-danger btn-confirmation', data: { placement: 'bottom' }) do
+          "<i class='fa fa-times' aria-hidden='true'></i> Delete".html_safe
+        end
+        [
+          view_context.link_to(timeline.title, timeline_path(timeline), title: timeline.description, target: "blank"),
+          timeline.description,
+          view_context.timeline_human_friendly_visibility(timeline.visibility),
+          "<span title='#{timeline.updated_at.utc.iso8601}'>#{view_context.time_ago_in_words(timeline.updated_at)} ago</span>",
+          timeline.tags.present? ? timeline.tags.join(', ') : '',
+          "#{copy_button} #{edit_button} #{delete_button}"
+        ]
+      end
+    }
+    respond_to do |format|
+      format.json do
+        render json: response
+      end
+    end
+  end
+
+  # GET /timelines/1
+  # GET /timelines/1.json
+  def show
+    authorize! :read, @timeline
+    # TODO: redirect to timeliner tool
+    redirect_to Settings.timeliner.timeliner_url #+ "?manifest=#{timeline_manifest_url(@timeline)}"
```

Comment:
Is the commented part going to be used later? If so we may need a short description on why/how.

Reply:
I can remove it but I'm not sure what the interface is yet for supplying a manifest url to the timeliner tool.